### PR TITLE
feat: add javadoc for a few methods used to run "fs ls" in Alluxio

### PR DESCRIPTION
**Pull request by abenaru-zz**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: [277](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/277)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [ ] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [ ] Checked if there is no possible merge conflicts before reviewing the PR
- [ ] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [ ] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [ ] Checked that code coverage for this PR is higher than 80%.
- [ ] Checked that there are no problems and code smells identified by the quality and security tools
- [ ] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [ ] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [ ] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: core/common/src/main/java/alluxio/conf/AlluxioConfiguration.java [hunk 1]
```diff
@@ -35,12 +35,15 @@
...
-   * @param options options for getting configuration value
-   * @return the value for the given key
+   * Gets the value for the given key in the properties of this configuration.
+   * <p>
+   * Implementations should return the value mapped to the provided {@code key}
+   * with the desired configuration value {@code options}.
+   *
+   * @param key the {@link PropertyKey} to get the value from
+   * @param options the {@link ConfigurationValueOptions} for getting the
+   *                configuration value from the {@code key}
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/conf/AlluxioConfiguration.java]: The second line of text should align with the first argument after the tag.

References: [https://github.com/Alluxio/alluxio/pull/11797/files](https://github.com/Alluxio/alluxio/pull/11797/files)

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 2]
```diff
@@ -2078,11 +2078,12 @@ public long createDirectory(AlluxioURI path, CreateDirectoryContext context)
...
-   * Implementation of directory creation for a given path.
+   * Effectively creates a directory for the provided inode path.
    *
-   * @param rpcContext the rpc context
-   * @param inodePath the path of the directory
-   * @param context method context
+   * @param rpcContext the {@link RpcContext} for establishing an RPC
+   * @param inodePath the {@link LockedInodePath} of the directory
+   * @param context the method {@link CreateDirectoryContext} used to merge
+   *                and wrap the options for creating the directory
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: Please watch the new standard of text alignment.

File: core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java [hunk 3]
```diff
@@ -193,17 +193,20 @@ private void checkOwner(LockedInodePath inodePath)
...
-   * @param path the path to check permission on
-   * @param inodeList file info list of all the inodes retrieved by traversing the path
-   * @param checkIsOwner indicates whether to check the user is the owner of the path
+   * @param user the user who requests access permission
+   * @param groups the groups the {@code user} belongs to
+   * @param bits bits that capture the action {@link Mode.Bits} by {@code user}
+   * @param path the path to check permissions from
+   * @param inodeList file info list of all the inodes retrieved by traversing the {@code path}
+   * @param checkIsOwner a boolean that indicates whether to check if the user is the owner of the
+   *                     path ({@code true}) or not ({@code false})
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/DefaultPermissionChecker.java]: Please watch the new standard of text alignment.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java [hunk 4]
```diff
@@ -389,6 +389,21 @@ private void applySetAcl(SetAclEntry entry) {
...
+   * in the {@link #mInodeStore}, updates it based on the provided
+   * {@link UpdateInodeEntry} and rewrites it to the inode store.
+   * <p>
+   * Persists the newly added inode if its persistence state is
+   * {@link PersistenceState#TO_BE_PERSISTED}.
+   *
+   * @param entry the {@link UpdateInodeEntry} to apply to the
+   *              inode
+   * @throws IllegalStateException If the update inode entry ID does not
+   *                               lead to an existing {@link InodeView}
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java]: Please watch the new standard of text alignment.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java [hunk 5]
```diff
@@ -389,6 +389,21 @@ private void applySetAcl(SetAclEntry entry) {
...
+   * <p>
+   * Gets the {@link MutableInode} for the provided {@code entry} ID
+   * in the {@link #mInodeStore}, updates it based on the provided
+   * {@link UpdateInodeEntry} and rewrites it to the inode store.
+   * <p>
+   * Persists the newly added inode if its persistence state is
+   * {@link PersistenceState#TO_BE_PERSISTED}.
+   *
+   * @param entry the {@link UpdateInodeEntry} to apply to the
+   *              inode
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java]: Please watch the new standard of text alignment.

File: core/server/master/src/main/java/alluxio/master/file/meta/MutableInodeDirectory.java [hunk 6]
```diff
@@ -79,8 +79,12 @@ public DefaultAccessControlList getDefaultACL() {
   }
 
   /**
-   * @param mountPoint the mount point flag value to use
-   * @return the updated object
+   * Determines whether this mutable inode directory is a mount point.
+   *
+   * @param mountPoint the mount point flag value to use, a boolean value
+   *                   representing whether this {@link InodeView} is a
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/meta/MutableInodeDirectory.java]: Please watch the new standard of text alignment.

**Review by caiopavanelli — CHANGES_REQUESTED** [review]

**Review by caiopavanelli — COMMENTED** [review]
Github is pointing some conflicts. Besides that it LGTM.
